code/modules/chemistry/tools/food_and_drink.dm

```diff
@@ -1392,6 +1387,43 @@
 	initial_volume = 120
 	shard_amt = 2
 
+/obj/item/reagent_containers/food/drinks/drinkingglass/icing
+	name = "icing tube"
+	desc = "Used to put icing on cakes."
+	icon = 'icons/obj/foodNdrink/food.dmi'
+	icon_state = "icing_tube"
+	initial_volume = 50
+	amount_per_transfer_from_this = 5
+	rc_flags = RC_FULLNESS | RC_VISIBLE | RC_SPECTRO
+
+	on_reagent_change()
+		src.underlays = null
+		if (reagents.total_volume >= 0)
+			if(reagents.total_volume == 0)
+				src.icon_state = "icing_tube"
+			else
+				src.icon_state = "icing_tube_2"
+			var/datum/color/average = reagents.get_average_color()
+			var/image/chem = new /image('icons/obj/foodNdrink/food.dmi',"icing_tube_chem")
+			chem.color = average.to_rgba()
+			src.underlays += chem
+		signal_event("icon_updated")
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		return
+
+	attack_self(var/mob/user as mob)
+		return
```

Comment:
This prevents players from from tightening and loosening their grip on the reagent container, is that intended?

Reply:
yesyes